packages/wrangler/src/pages/deploy.tsx

```diff
@@ -73,38 +75,65 @@ export function Options(yargs: CommonYargsArgv) {
 }
 
-export const Handler = async ({
-	_,
-	directory,
-	projectName,
-	branch,
-	commitHash,
-	commitMessage,
-	commitDirty,
-	skipCaching,
-	bundle,
-	noBundle,
-	config: wranglerConfig,
-}: PublishArgs) => {
+export const Handler = async (args: PagesDeployArgs) => {
+	let { branch, commitHash, commitMessage, commitDirty } = args;
+
 	// Check for deprecated `wrangler pages publish` command
-	if (_[1] === "publish") {
+	if (args._[1] === "publish") {
 		logger.warn(
 			"`wrangler pages publish` is deprecated and will be removed in the next major version.\nPlease use `wrangler pages deploy` instead, which accepts exactly the same arguments."
 		);
 	}
 
-	if (wranglerConfig) {
-		throw new FatalError("Pages does not support wrangler.toml", 1);
+	if (args.config) {
+		throw new FatalError(
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
+	}
+
+	/*
+	 * Check whether project contains a valid Pages `wrangler.toml` file. If it
+	 * does, we want to pick up the configuration specified in it. If it doesn't
+	 * we'll ignore the file, but inform users that we did find one, just not
+	 * valid for Pages.
+	 */
+	let config: Config | undefined;
+	const configPath = findWranglerToml(process.cwd(), false);
+	if (configPath) {
+		// this reads the config file with `env` set to `undefined`, which will
+		// return the top-level config. This contains all the information we
+		// need from now. We will perform a second config file read later
+		// in `/api/pages/deploy`, that will get the environment specififc config
+		config = readConfig(configPath, { ...args, env: undefined });
+		if (config.pages_build_output_dir === undefined) {
+			logger.warn(
+				`Pages now has \`wrangler.toml\` support.\n` +
+					`We detected a configuration file at ${config.configPath} but it is missing the "pages_build_output_dir" field, required by Pages.\n` +
+					`If you would like to use this configuration file to deploy your project, please use "pages_build_output_dir" to specify the directory of static files to upload.\n` +
+					`Ignoring configuration file for now, and proceeding with project deploy.`
+			);
+
+			// set to `undefined` so we can ignore it from here on
+			config = undefined;
+		}
 	}
 
+	const directory = args.directory ?? config?.pages_build_output_dir;
 	if (!directory) {
-		throw new FatalError("Must specify a directory.", 1);
+		throw new FatalError(
+			"Must specify a directory of assets to deploy. Please specify the [<directory>] argument in the `pages deploy` command, or configure `pages_build_output_dir` in your `wrangler.toml` configuration file.",
+			1
+		);
 	}
 
-	const config = getConfigCache<PagesConfigCache>(PAGES_CONFIG_CACHE_FILENAME);
-	const accountId = await requireAuth(config);
+	const configCache = getConfigCache<PagesConfigCache>(
+		PAGES_CONFIG_CACHE_FILENAME
+	);
+	const accountId = await requireAuth(configCache);
 
-	projectName ??= config.project_name;
+	let projectName =
```

Comment:
// TODO @CarmenPopoviciu 
if the project specified by `wrangler.toml`'s `name` does not exist, `pages deploy` will fail, because we're attempting to get the project [here](https://github.com/cloudflare/workers-sdk/blob/main/packages/wrangler/src/api/pages/deploy.tsx#L134). In combination with the `name` field being mandatory, and Pages toml validation throwing an err if field is missing, this basically puts us in the position where we can not deploy **new** Pages projects with a `wrangler.toml` file, only **existing** ones.

We should instead warn that the project doesn't exist and offer users the option of us creating the project for them, like we for projects without a config file

Reply:
fixed, YAY 🎉 😍 ! Will add some comprehensive tests tomorrow and link here

Reply:
turns out writing tests for interactive pages deploy is not as trivial as I thought, mainly because we're doing some funky prompting that is hard to mock. Will address this as a followup